FlashCap.Core/Internal/NativeMethods_DirectShow.cs

```diff
@@ -297,6 +297,54 @@ public struct FILTER_INFO
         public IFilterGraph graph;
     }
 
+
+    [StructLayout(LayoutKind.Sequential)]
+    public struct DsCAUUID
+    {
+        public int cElems;
+        public IntPtr pElems;
+
+        public Guid[] ToGuidArray()
+        {
+            Guid[] retval = new Guid[cElems];
+
+            for (int i = 0; i < cElems; i++)
+            {
+#pragma warning disable CS0618
```

Comment:
I can't find a way to call the `Marshal.<T>` ones

Reply:
I am extremely late in responding due to a lot of private work. My apologies 😵‍💫

Is it correct that what you want to do here is to copy the GUID structure pointed to by pElems into an array and return it? If so, how about this code for example?

```csharp
        public unsafe Guid[] ToGuidArray()
        {
            var retval = new Guid[cElems];
            fixed (Guid* pr = &retval[0])
            {
                Guid* pd = pr;
                Guid* ps = (Guid*)pElems;
                for (var index = 0; index < this.cElems; index++)
                {
                    *pd = *ps;
                    ps++;
                    pd++;
                }
            }
            return retval;
        }
```

* We can use the unsafe block in FlashCap, It is used it a lot. (although care must be taken in implementation)
* There should have been compatibility between the layout of the GUID structure in native code and the internal layout of the Guid structure in .